central/auth/service/service_impl_test.go

```diff
@@ -126,3 +181,428 @@ func (s *authServiceAccessControlTestSuite) TestAuthServiceResponse() {
+				IssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuerConfig{
+					GenericIssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuer{
+						Issuer: "https://stackrox.io",
+					},
+				},
+				Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole1,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole2,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole3,
+					},
+				},
+			},
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "5m",
+				Type:                    v1.AuthMachineToMachineConfig_GENERIC,
+				IssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuerConfig{
+					GenericIssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuer{
+						Issuer: "https://stackrox.io",
+					},
+				},
+				Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole1,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole2,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole3,
+					},
+				},
+			},
+		},
+		{
+			skipIDCheck: true,
+			config: &v1.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "5m",
+				Type:                    v1.AuthMachineToMachineConfig_GENERIC,
+				IssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuerConfig{
+					GenericIssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuer{
+						Issuer: "https://stackrox.io",
+					},
+				},
+				Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole1,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole2,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole3,
+					},
+				},
+			},
+		},
+	}
+
+	for i, testCase := range testCases {
+		s.Run(fmt.Sprintf("test case %d", i), func() {
+			err := s.svc.validateAuthMachineToMachineConfig(sac.WithAllAccess(context.Background()),
+				testCase.config, testCase.skipIDCheck)
+			s.ErrorIs(err, testCase.err)
+		})
+	}
+}
+
+func (s *authServiceAccessControlTestSuite) TestGetConfig() {
+	addConfigResp, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "something",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "github",
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	id := uuid.NewV4().String()
+	fmt.Print(id)
```

Comment:
Did you forget to plug this into further code?

Reply:
ah this should have been removed, I removed it.